src/features/user/Settings/EditProfile/AddressManagement/AddAddress.tsx

```diff
@@ -70,8 +73,25 @@ const AddAddress = ({
           token,
           logoutUser,
         });
-        if (res && setUserAddresses) {
-          setUserAddresses((prevAddresses) => [...prevAddresses, res]);
+        if (res) {
+          setUser((prev) => {
+            if (!prev) return null;
+
+            const updatedAddresses =
+              res.type === ADDRESS_TYPE.PRIMARY
+                ? prev.addresses.map((addr) =>
+                    addr.type === ADDRESS_TYPE.PRIMARY
+                      ? { ...addr, type: ADDRESS_TYPE.OTHER }
+                      : addr
+                  )
+                : prev.addresses;
+
+            return {
+              ...prev,
+              addresses: [...updatedAddresses, res],
+            };
+          });
+          updateUserAddresses();
```

Comment:
After updating the user object, is `updateUserAddresses()` actually needed? The user object will contain the updated addresses.

Reply:
While this has been made optional, I don't think its needed for src/features/user/Settings/EditProfile/AddressManagement/index.tsx as well.

Reply:
Created a [issue](https://github.com/Plant-for-the-Planet-org/planet-webapp/issues/2417) it